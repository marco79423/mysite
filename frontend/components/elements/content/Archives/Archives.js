import React from 'react'
import dateformat from 'dateformat'
import styled from 'styled-components'

import {Table, Tbody, Td, Tr} from '../../generic/table'
import NormalLink from '../../generic/NormalLink'
import TitleLink from '../../generic/TitleLink'
import Loading from '../../generic/Loading'


const Base = styled.section`
  float: left;
  width: 800px;

  @media (max-width: 1200px) {
    width: 100%;
  }

  article {
    padding: 2em;
    border-bottom: 1px solid #eee;
    background: white;

    table {
      
    }
  }
`

const Header = styled.header`
  h1 {
    margin: 3px 0 24px;
  }
`

export default class Archives extends React.PureComponent {

  renderHeader = () => {
    return (
      <Header>
        <h1><TitleLink href='/articles/archives/'>所有文章列表</TitleLink></h1>
      </Header>
    )
  }

  renderArticleTable = () => {
    const {articles} = this.props
    return (
      <Table>
        <Tbody>
        {articles.map(article => (
          <Tr key={article.slug}>
            <Td>{dateformat(article.date, 'yyyy/mm/dd')}</Td>
            <Td><NormalLink href={`/articles/${article.slug}/`}>{article.title}</NormalLink></Td>
          </Tr>
        ))}
        </Tbody>
      </Table>
    )
  }

  render() {
    const {articles} = this.props
    if (articles.length === 0) {
      return (
        <Base>
          <Loading/>
        </Base>
      )
    }

    return (
      <Base>
        <article>
          {this.renderHeader()}
          {this.renderArticleTable()}
        </article>
      </Base>
    )
  }
}
