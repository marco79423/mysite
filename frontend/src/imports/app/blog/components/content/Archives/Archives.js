import React from 'react'
import PropTypes from 'prop-types'
import ImmutablePropTypes from 'react-immutable-proptypes'
import dateformat from 'dateformat'
import styled from 'styled-components'

import {Table, Tbody, Tr, Td} from '../../generic/table'
import NormalLink from '../../generic/NormalLink'
import TitleLink from '../../generic/TitleLink'
import Loading from '../../generic/Loading'


const Base = styled.section`
  float: left;
  width: 800px;

  @media (max-width: 1200px) {
    width: 100%;
  }

  article {
    padding: 2em;
    border-bottom: 1px solid #eee;
    background: white;

    table {
      
    }
  }
`

const Header = styled.header`
  h1 {
    margin: 3px 0 24px;
  }
`

export default class Archives extends React.PureComponent {
  static PropTypes = {
    articles: ImmutablePropTypes.listOf(
      ImmutablePropTypes.contains({
        slug: PropTypes.string.isRequired,
        title: PropTypes.string.isRequired,
        date: PropTypes.any.isRequired
      })
    )
  }

  renderHeader = () => {
    return (
      <Header>
        <h1><TitleLink to='/articles/archives/'>所有文章列表</TitleLink></h1>
      </Header>
    )
  }

  renderArticleTable = () => {
    const {articles} = this.props
    return (
      <Table>
        <Tbody>
        {articles.map(article => (
          <Tr key={article.get('slug')}>
            <Td>{dateformat(article.get('date'), 'yyyy/mm/dd')}</Td>
            <Td><NormalLink to={`/articles/${article.get('slug')}/`}>{article.get('title')}</NormalLink></Td>
          </Tr>
        ))}
        </Tbody>
      </Table>
    )
  }

  render () {
    const {articles} = this.props
    if (articles.isEmpty()) {
      return (
        <Base>
          <Loading/>
        </Base>
      )
    }

    return (
      <Base>
        <article>
          {this.renderHeader()}
          {this.renderArticleTable()}
        </article>
      </Base>
    )
  }
}
