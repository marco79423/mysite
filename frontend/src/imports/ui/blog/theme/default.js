const colorTable = {
  'cyan-50': '#E0F7FA',
  'cyan-600': '#00ACC1',
  'cyan-700': '#0097A7',
  'cyan-800': '#00838F',
  'cyan-900': '#006064',
  'cyan-A700': '#00B8D4',

  'teal-50':'#E0F2F1',
  'teal-100':'#B2DFDB',
  'teal-200':'#80CBC4',
  'teal-300':'#4DB6AC',
  'teal-400':'#26A69A',
  'teal-500':'#009688',
  'teal-600':'#00897B',
  'teal-700':'#00796B',
  'teal-800':'#00695C',
  'teal-900':'#004D40',
  'teal-A700':'#00BFA5',

  'grey-50': '#FAFAFA',
  'grey-200': '#EEEEEE',
  'grey-300': '#E0E0E0',
  'grey-600': '#757575',
  'grey-700': '#616161',
  'grey-900': '#212121'
}

export default {
  global: {
    fontColor: colorTable['grey-900'],
    link: {
      color: colorTable['cyan-700'],
      hoverColor: colorTable['cyan-900']
    }
  },
  page: {
    background: colorTable['grey-700'],
    header: {
      background: colorTable['teal-400'],
      titleColor: colorTable['grey-50']
    },
    nav: {
      background: colorTable['teal-600'],
      menu: {
        color: '#ECECEC',
        background: colorTable['teal-600'],
        hoverBackground: colorTable['cyan-900'],
        dropdown: {
          background: colorTable['teal-600'],
          borderColor: colorTable['teal-800']
        }
      }
    },
    main: {
      background: colorTable['grey-200'],
      content: {
        article: {
          background: 'white',
          metadata: {
            tip: {
              color: '#FFFFFF',
              background: colorTable['teal-600']
            }
          },
          header: {
            color: colorTable['grey-700']
          },
          block: {
            color: 'white',
            background: colorTable['teal-600'],
            borderColor: '#ccc'
          },
          doctestBlock: {
            color: 'white',
            background: '#6D6D6D',
          },
          note: {
            background: '#ECECEC',
            titleColor: colorTable['teal-800'],
            borderColor: '#ccc'
          },
          table: {
            header: {
              background: colorTable['teal-600'],
            }
          }
        },
        pagination: {
          color: colorTable['grey-200'],
          background: colorTable['teal-700'],
          hoverBackground: colorTable['teal-900'],
          activeBackground: colorTable['teal-900'],
        }
      },
      aside: {
        background: colorTable['grey-200'],
        darkerBackground: colorTable['grey-600'],

        section: {
          titleColor: colorTable['grey-700'],
          borderColor: colorTable['grey-200']
        }
      }
    },
    footer: {
      background: colorTable['teal-600'],
      copyrightColor: '#ECECEC',
    }
  }
}