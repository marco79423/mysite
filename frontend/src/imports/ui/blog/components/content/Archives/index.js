import Archives from './Archives'

export default Archives
