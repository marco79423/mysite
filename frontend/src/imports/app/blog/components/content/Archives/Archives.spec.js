import Immutable from 'immutable'
import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import {ThemeProvider} from 'styled-components'

import Archives from './Archives'
import theme from '../../../theme/default'

test('It should render Archives component with empty articles correctly', () => {
  const articles = Immutable.fromJS([])

  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <Archives articles={articles}/>
    </ThemeProvider>
  )
  expect(component.toJSON()).toMatchSnapshot()
})

test('It should render Archives component correctly', () => {
  const articles = Immutable.fromJS([
    {slug: 'slug 1', title: 'title 1', date: new Date(2017, 4, 15)},
    {slug: 'slug 2', title: 'title 2', date: new Date(2017, 4, 16)},
    {slug: 'slug 3', title: 'title 3', date: new Date(2017, 4, 17)}
  ])

  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <Archives articles={articles}/>
    </ThemeProvider>
  )
  expect(component.toJSON()).toMatchSnapshot()
})
