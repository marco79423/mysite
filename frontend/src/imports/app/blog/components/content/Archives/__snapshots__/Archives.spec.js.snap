// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render Archives component correctly 1`] = `
.c4 {
  border-collapse: collapse;
  width: 100%;
  font-size: 1.2em;
}

.c5 {
  background: #eee;
}

.c5:nth-of-type(even) {
  background: #f9f9f9;
}

.c6 {
  padding: 0.6em;
  border: 1px solid #ccc;
}

.c3 {
  color: #0097A7;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3:hover {
  color: #006064;
}

.c2 {
  font-size: 2rem;
}

.c0 {
  float: left;
  width: 800px;
}

.c0 article {
  padding: 2em;
  border-bottom: 1px solid #eee;
  background: white;
}

.c1 h1 {
  margin: 3px 0 24px;
}

@media (max-width:1200px) {
  .c0 {
    width: 100%;
  }
}

<section
  className="c0"
>
  <article>
    <header
      className="c1"
    >
      <h1>
        <a
          className="c2 c3"
          onClick={[Function]}
          style={Object {}}
        >
          所有文章列表
        </a>
      </h1>
    </header>
    <table
      className="c4"
    >
      <tbody
        className=""
      >
        <tr
          className="c5"
        >
          <td
            className="c6"
          >
            2017/05/15
          </td>
          <td
            className="c6"
          >
            <a
              className="c3"
              onClick={[Function]}
              style={Object {}}
            >
              title 1
            </a>
          </td>
        </tr>
        <tr
          className="c5"
        >
          <td
            className="c6"
          >
            2017/05/16
          </td>
          <td
            className="c6"
          >
            <a
              className="c3"
              onClick={[Function]}
              style={Object {}}
            >
              title 2
            </a>
          </td>
        </tr>
        <tr
          className="c5"
        >
          <td
            className="c6"
          >
            2017/05/17
          </td>
          <td
            className="c6"
          >
            <a
              className="c3"
              onClick={[Function]}
              style={Object {}}
            >
              title 3
            </a>
          </td>
        </tr>
      </tbody>
    </table>
  </article>
</section>
`;

exports[`It should render Archives component with empty articles correctly 1`] = `
.c1 {
  margin-top: 10%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  display: inline-block;
  width: 90px;
  height: 90px;
  background: url() no-repeat center;
  background-size: 90px 90px;
  border: 0.1em solid #FFFFFF;
  box-shadow: 0 1px 4px rgba(0,0,0,0.15);
  -webkit-animation-name: bQMHby;
  animation-name: bQMHby;
  -webkit-animation-duration: 5s;
  animation-duration: 5s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  z-index: 1000;
}

.c3 {
  margin-top: 2rem;
  font-size: 2rem;
  text-align: center;
}

.c0 {
  float: left;
  width: 800px;
}

.c0 article {
  padding: 2em;
  border-bottom: 1px solid #eee;
  background: white;
}

@media (max-width:1200px) {
  .c0 {
    width: 100%;
  }
}

<section
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    />
    <div
      className="c3"
    >
      正在努力讀取中…
    </div>
  </div>
</section>
`;
